lib/Agrammon/Web/Service.pm6

```diff
@@ -271,14 +272,27 @@ class Agrammon::Web::Service {
         $model-path.IO.&child-secure($technical).slurp
     }
 
-    method get-outputs-from-csv(
-        Agrammon::DB::User $user,
-        Str $simulation-name, Str $dataset-name, $csv-data,
+    subset InputFormats of Str where  { $_ eq 'application/json' or $_ eq 'text/csv' };
+    subset OutputFormats of Str where { $_ eq 'application/json' or $_ eq 'text/csv' or $_ eq 'text/plain' };
+
+    #| Run model from input data as CSV or JSON
+    #| and return output formatted as CSV, JSON, or TEXT
+    method get-outputs-for-rest(
+        Str $simulation-name, Str $dataset-name, $input-data, InputFormats $type,
         :$model-version, :$variants, :$technical-file,
-        :$language, :$format, :$print-only,
+        :$language, OutputFormats :$format!, :$print-only,
         :$include-filters, :$all-filters
     ) {
-        my $input = Agrammon::DataSource::CSV.new.from-csv($simulation-name, $dataset-name, $csv-data);
+        my $data-source;
+        given $type {
+            when 'text/csv' {
+                $data-source = Agrammon::DataSource::CSV.new;
+            }
+            when 'application/json' {
+                $data-source = Agrammon::DataSource::JSON.new;
+            }
```

Comment:
Maybe a `default` just in case something else slips in?

Reply:
`$type` is typed as `subset InputFormats` not allowing anything else in. And the OpenAPI definition doesn't allow anything else either. Would you still recommend a default?

Reply:
I was just thinking a simple `die` (doesn't even have to be typed) just in case. But yeah, I agree it can't be reached at present.

Reply:
Ok, added a `die` just in case somebody drops the checks :-)